pkg/gardenlet/controller/networkpolicy/controller.go

```diff
@@ -155,6 +155,7 @@ func (c *Controller) Run(ctx context.Context, workers int) {
 			return policy.Name == helper.AllowToSeedAPIServer
 		},
 		Handler: cache.ResourceEventHandlerFuncs{
+			AddFunc:    c.networkPolicyAdd,
```

Comment:
Can you explain the motivation for this change?

Reply:
The reconciler deletes the `allow-to-seed-apiserver` NetworkPolicy in namespaces which are not shoot, gardener or istio-system namespaces. If the reconciler does not watch add events, it is unpredictable when this happens in case someone creates a policy. It might happen on namespace changes, on controller restart (because of namespace add event), on NetworkPolicy change, on Kubernetes endpoint change.
However, I'm not really sure why the controller should delete the policy in the first place. If this is needed, it should do in a predictable way.